[redesign][ln] Receive tab

bgptr opened the pull request:
Toward issue #3296

Some notes:
- show settle date on invoice modal instead of expiration time when the payment already has been received
- cancel invoice feature added
- add snackbar messages for indicating add and cancel invoice events
- implemented filter, sort, and search function for invoice list
- tests added

Updated screenshots:
<img width="674" alt="DeepinScreenshot_select-area_20210809215032" src="https://user-images.githubusercontent.com/52497040/128766453-76ab661e-5b7e-483c-bf27-22f859a8ac5f.png">
<img width="585" alt="DeepinScreenshot_select-area_20210805140422" src="https://user-images.githubusercontent.com/52497040/128347005-26acbe04-0c19-47ba-83bc-f2d537fdcd81.png">
<img width="585" alt="DeepinScreenshot_select-area_20210809214956" src="https://user-images.githubusercontent.com/52497040/128766470-aebd147b-445a-4301-84f0-a2b6b83bbe1c.png">

matheusd reviewed (commented):
This is going in the right track, but I see some UX issues:

- You can't create a 0 amount invoice anymore. This is needed for "donations" (a 0 amount invoice indicates the sender can send any amount)
- Both canceled and timed out invoices are appearing as "canceled" (not sure that's the intention)
- Maybe we should disable copying the payment request if the payment was canceled/timed out? It will fail if someone attempts to pay for it, so this could make it less prone to cause this error
- Amount in the individual invoice box (in the invoices listing) is showing a "+" sign
- I'd like to get an additional expandable view (either in the details modal or directly in the listing) with the extended info for the invoice (hash, description, htlcs, etc).

bgptr commented:
@matheusd Thanks for your review. I've pushed the suggested changes and updated the screenshots.

bgptr commented:
Update: implemented the polished details part of the invoice modal thanks to @MariaPleshkova. (screenshots are updated)
Also, converted the details table to a shared component since more of the ln pages will use it.

matheusd commented on app/components/modals/LNInvoiceModal/helpers.js:
```diff
@@ -0,0 +1,75 @@
+  const details = [
+    {
+      label: <T id="ln.invoicesModal.hash" m="Hash" />,
+      value: invoice?.rHash
```
This is being displayed as a base64 encoding. Usually this is shown as an hex.

matheusd reviewed (commented):
Also, it's probably useful to have the payment hash be copyable, since that's used in the command line to verify things.

MariaPleshkova commented:
@bgptr 
I added the copy button next to the hash as @matheusd suggested. We already use the same button for the Remote Pubkey in the Channel Properties.
![image](https://user-images.githubusercontent.com/8841018/128535846-e883ddf6-2f34-4238-94fc-67ca7240fcb3.png)

bgptr commented:
@matheusd @MariaPleshkova Thanks, guys. I pushed the changes and updated the screenshots.

matheusd reviewed (approved):
Working nicely now!
